### ISSUE TITLE
Add Linux Support

### ISSUE THREAD

**Issue by mgntf4n6od** [microsoft/libyara.NET#25]
.NET 6 is the latest LTS supported version of .NET at the time of writing. Are there any plans to release a version of this library that targets NET 6+ and supports Linux? 

Targetting .NET 6+ might simplify dependency choices for new consumers of this library moving forward (currently, there's a .NET Core and .NET Framework version of the nuget package to select between)

**Comment by ApoIIy0n** [comment]
No news so far?

**Comment by swannman** [comment]
@ApoIIy0n or @mgntf4n6od are you experiencing issues when libyara.NET is imported into a project that targets .NET 6?

**Comment by ApoIIy0n** [comment]
@swannman Yes mate, the moment I implement it and build it as a linux-x64 build , the libyara.NET.dll will be in a incorrect format
and yes it works flawless on Windows only

**Comment by ApoIIy0n** [comment]
@swannman little update, I think I'm doing everything correct on my side now (made a new project, checked all settings), though the moment I load the libyaraNET within the application I get a segmentation fault on Linux.

**Comment by kallanreed** [comment]
This only targets Windows. We use C++ CLI and a Windows build of libyara.

On Jan 12, 2023, at 00:03, Apollyon ***@***.***> wrote:

﻿

@swannman<https://github.com/swannman> little update, I think I'm doing everything correct on my side now (made a new project, checked all settings), though the moment I load the libyaraNET within the application I get a segmentation fault on Linux.

**Comment by ApoIIy0n** [comment]
@kallanreed hence the reply from me on the OP in request for full support for cross-platform, because @swannman asked me if one was experiencing issues, I simply replied what I came across while trying it on Linux.

**Comment by kallanreed** [comment]
I’m simply explaining why you get a seg fault on Linux. It’s because you’re trying to run a Windows binary.

This issue should be renamed, “Add Linux Support”, because that is orthogonal to whether or not we support targeting .NET 6.

**Comment by ApoIIy0n** [comment]
@kallanreed roger, I hope the OP actually needed it for Linux as well then. Thanks so far, now hope it's actually going to be considered.

**Comment by mgntf4n6od** [comment]
I've updated the description and title to reflect that. 

+1 request to have this work on Linux, if possible.

I _think_ that would mean using something other than C++/CLI to bridge into the native YARA stuff. Perhaps PIvokes is the cross-platform way to do this? 
This would also need a linux binary of the native library wrapped here.

**Comment by kallanreed** [comment]
Just a few notes for whoever looks into this:
1) need to check upstream and see if the yara project ever moved to using CMake. If they haven’t, we should help with that.
2) We had been using a pre-built, static lib of libyara that I packaged as a nuget. The best thing to do would be to use the source release and build it as part of building this. This becomes a lot easier if upstream is using CMake.
3) Yeah, we basically would have to write all the pInvoke structures and then wrap the C calls instead of using C++CLI. Also a good move because MSVC’s commitment to CLI support is “begrudgingly” at best. Honestly, getting it all building with Clang would make the X-plat story a lot easier.  Victor might have already done this upstream?
